modulus/datapipes/climate/utils/invariant.py

```diff
@@ -0,0 +1,119 @@
+
+    # get latitudes and longitudes from data shape
+    lat = np.linspace(*bounds[0], shape[0], dtype=np.float32)
+
+    # does longitude wrap around the globe?
+    lon_wraparound = ((bounds[1][0] % 360) == (bounds[1][1] % 360))
+    if lon_wraparound:
+        # treat differently from lat due to wrap-around
+        lon = np.linspace(*bounds[1], shape[1]+1, dtype=np.float32)[:-1]
+    else:
+        lon = np.linspace(*bounds[1], shape[1], dtype=np.float32)
+
+    return np.meshgrid(lat, lon, indexing='ij')
+
+
+class Invariant(ABC):
+    @abstractmethod
+    def __call__(self, latlon: np.ndarray):
+        pass
+
+
+class LatLon(Invariant):
+    def __init__(
+        self,
+        outputs: List[str] = ("sin_lat", "cos_lat", "sin_lon", "cos_lon")
+    ):
+        """
+        Outputs latitude and longitude and their trigonometric functions.
+
+        Parameters
+        ----------
+        outputs: List[str]
+            List of outputs. Supported values are
+            `{"lat", "lon", "sin_lat", "cos_lat", "sin_lon", "cos_lon"}`
+        """
+        self.outputs = outputs
+
+    def __call__(self, latlon: np.ndarray):
+        (lat, lon) = latlon
+
+        vars = {"lat": lat, "lon": lon}
+
+        # cos/sin latitudes and longitudes
+        if "sin_lat" in self.outputs:
+            vars["sin_lat"] = np.sin(np.deg2rad(lat))
+        if "cos_lat" in self.outputs:
+            vars["cos_lat"] = np.cos(np.deg2rad(lat))
+        if "sin_lon" in self.outputs:
+            vars["sin_lon"] = np.sin(np.deg2rad(lon))
+        if "cos_lon" in self.outputs:
+            vars["cos_lon"] = np.cos(np.deg2rad(lon))
+
+        return np.stack([vars[o] for o in self.outputs], axis=0)
+
+
+class FileInvariant(Invariant):
+    def __init__(
+        self,
+        filename: str,
+        var_name: str,
+        normalize=False,
+        interp_method='linear',
+    ):
+        """
+        Loads an time-invariant variable from a NetCDF4 file. The file should
+        contain one or more data variables of dimensions 
+        `(channels, latitude, longitude)` as well as variables `latitude` and
+        `longitude` specifying these coordinates. `latitude` and `longitude`
+        can be either 2D or 1D.
+
+        Parameters
+        ----------
+        filename: str
+            Path to the file containing the variable
+        var_name: str
+            The variable in the file containing the data
+        normalize: bool, optional
+            If True, normalize the data by to zero-mean and unit variance.
+            Default False.
+        interp_method: str, optional
+            Any argument accepted by xarray.DataArray.interp.
+            Default 'linear'.
+        """
+
+        with xr.open_dataset(filename) as ds:
+            self.data = ds[var_name].astype(np.float32)
+            self.lat = ds["latitude"].to_numpy().astype(np.float32)
+            self.lon = ds["longitude"].to_numpy().astype(np.float32)
+
```

Comment:
So what if I build a NetCDF4 file with 'lat' and 'lon' instead of 'latitude' and 'longitude'?

Reply:
What if I have more than 1 `var_name` that I want to load?

Reply:
> So what if I build a NetCDF4 file with 'lat' and 'lon' instead of 'latitude' and 'longitude'?
Sure we could have the names configurable but is it worth the added complexity? After the all the user can always rename the NetCDF variables, or build their own function/class to load from whatever kind of file they want.

Reply:
I suppose not, this function is well documented with variable assumptions in any case.

Reply:
I think my question about the single `var_name` remains: What is the decision for only allowing a single `var_name` and is it conceivable that someone would want to load additional invariants?